src/ansiblelint/config.py

```diff
@@ -164,5 +165,27 @@ def ansible_version(version: str = "") -> Version:
     return Version(version)
 
 
+def path_inject() -> None:
+    """Add python interpreter path to top of PATH to fix outside venv calling."""
+    # This make it possible to call ansible-lint that was installed inside a
+    # virtualenv without having to pre-activate it. Otherwise subprocess will
+    # either fail to find ansible executables or call the wrong ones.
+    #
+    # This must be run before we do run any subprocesses, and loading config
+    # does this as part of the ansible detection.
+    paths = [x for x in os.environ.get("PATH", "").split(os.pathsep) if x]
+    ansible_path = shutil.which("ansible")
+    if ansible_path:
+        ansible_path = os.path.dirname(ansible_path)
+    py_path = os.path.dirname(sys.executable)
+    # Determine if we need to manipulate PATH
+    if ansible_path not in paths or py_path != ansible_path:  # pragma: no cover
+        # tested by test_call_from_outside_venv but coverage cannot detect it
+        paths.insert(0, py_path)
+        os.environ["PATH"] = os.pathsep.join(paths)
+        print(f"WARNING: PATH altered to include {py_path}", file=sys.stderr)
+
+
+path_inject()
 if ansible_collections_path() in os.environ:
     collection_list = os.environ[ansible_collections_path()].split(":")
```

Comment:
@webknjaz The import-time side effects are here. `ansible_collections_path()` calls `ansible_version()` which runs `ansible --version` in a subprocess. So, running `path_inject()` in `main()` is too late. Changing things so that `collection_list` is not created at import time is out-of-scope for this PR.